apps/ejabberd/src/cyrsasl.erl

```diff
@@ -36,15 +36,12 @@
 
 -include("ejabberd.hrl").
 
--record(sasl_mechanism, {mechanism :: mechanism(),
+-record(sasl_mechanism, {mechanism :: mechanism() | atom(), % atom() used in ets:select
                          module :: sasl_module(),
-                         password_type :: plain | digest | scram
+                         password_type :: plain | digest | scram | atom() % atom() used in ets:select
                         }).
--type sasl_module() :: cyrsasl_anonymous
-                     | cyrsasl_digest
-                     | cyrsasl_plain.
+-type sasl_module() :: atom().
```

Comment:
I know the `cyrsasl_scram` was missing but only because we didn't care about specs before. In my  opinion it would be better to have all the possible cyrsasl modules defined in the type. But maybe some else has other opinion on this. Anyone? @lavrin? @pzel?